src/Symfony/Contracts/HttpClient/ApiClientInterface.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Contracts\HttpClient;
+
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+use Symfony\Contracts\HttpClient\Test\ApiClientTestCase;
+
+/**
+ * Provides flexible methods for interacting with HTTP APIs.
+ *
+ * Unless otherwise specified, implementations MUST send a JSON-compatible
+ * "accept" header - typically "application/json".
+ *
+ * @see HttpClientInterface for a description of options and how they MUST work
+ * @see ApiClientTestCase   for a reference test suite
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+interface ApiClientInterface
+{
+    const OPTIONS_DEFAULTS = HttpClientInterface::OPTIONS_DEFAULTS;
+
+    /**
+     * Requests an HTTP resource using the GET verb.
+     *
+     * @throws TransportExceptionInterface When an unsupported option is passed
+     */
+    public function get(string $url, iterable $options = []): ApiResponseInterface;
```

Comment:
these http method named methods would make sense with any client, not only for an API client. why is this tied into api client and not a decorator for the plain client?

Reply:
What I mean by "api" is that the response bodies are small enough to be buffered in memory. E.g. a `.iso` of a video file wouldn't fit this definition - one should use `HttpClientInterface` then.
A decorator wouldn't provide the required abstraction level: I'm looking for something ppl can type-hint for without binding them to one implementation.